qc_tests/streaks.py

```diff
@@ -93,102 +170,354 @@ def get_repeating_string_threshold(obs_var, config_dict, plots=False, diagnostic
+            config_dict[f"STREAK-{this_var.name}"] = CD_straight
+
+    return # get_excess_streak_threshold
+
 
 #************************************************************************
-def repeating_value(obs_var, times, config_dict, plots=False, diagnostics=False):
+def repeating_value(obs_var: utils.Meteorological_Variable, times: np.array,
+                    config_dict: dict,
+                    wind_speed: Optional[utils.Meteorological_Variable] = None,
+                    plots: bool = False, diagnostics: bool = False) -> None:
     """
-    AKA straight string
+    AKA straight streak
 
-    Use config file to read threshold values.  Then find strings which exceed threshold.
+    Use config file to read threshold values.  Then find streaks which exceed threshold.
 
     :param MetVar obs_var: meteorological variable object
     :param array times: array of times (usually in minutes)
     :param str config_dict: configuration file to store critical values
+    :param MetVar wind_speed: need speeds to mask calm periods in wind_directions
     :param bool plots: turn on plots
     :param bool diagnostics: turn on diagnostic output
     """
  
-    # remove calm periods for wind speeds when (a) calculating thresholds and (b) identifying streaks
-    this_var = copy.deepcopy(obs_var)
-    if obs_var.name == "wind_speed":
-        calms, = np.ma.where(this_var.data == 0)
-        this_var.data[calms] = utils.MDI
-        this_var.data.mask[calms] = True
-
-
+    this_var = prepare_obs_var(obs_var, wind_speed=wind_speed)
+ 
     flags = np.array(["" for i in range(this_var.data.shape[0])])
     compressed_flags = np.array(["" for i in range(this_var.data.compressed().shape[0])])
+    masked_times = np.ma.array(times, mask=obs_var.data.mask)
 
-    # retrieve the threshold and store in another dictionary
-    threshold = {}
+    # retrieve the threshold
     try:
-        th = config_dict[f"STREAK-{this_var.name}"]["Straight"]
-        threshold["Straight"] = th
+        threshold = config_dict[f"STREAK-{this_var.name}"]["Straight"]
     except KeyError:
         # no threshold set
-        get_repeating_string_threshold(this_var, config_dict, plots=plots, diagnostics=diagnostics)
-        th = config_dict[f"STREAK-{this_var.name}"]["Straight"]
-        threshold["Straight"] = th
+        get_repeating_streak_threshold(this_var, config_dict, plots=plots, diagnostics=diagnostics)
+        threshold = config_dict[f"STREAK-{this_var.name}"]["Straight"]
 
+    if threshold == utils.MDI:
+        # No threshold obtainable, no need to continue the test
+        return
+    
     # only process further if there is enough data
-    if len(this_var.data.compressed()) > 1:
-        repeated_string_lengths, grouped_diffs, strings = utils.prepare_data_repeating_string(this_var.data.compressed(), diff=0, plots=plots, diagnostics=diagnostics)
+    if len(this_var.data.compressed()) < 2:
+        # Escape if insufficient data
```

Comment:
Think I've already commented here that its not obvious what 'enough' data is. I'm not familiar with .compressed() to know why you would change from > 1 to < 2. So more commenting would be helpful here?

Reply:
I changed the logic - rather than lots of nested if-loops, I've changed where I could so that the subroutine returns if something isn't met.  This reduces the indenting.  So rather than "continue processing if data array has > 1 value", it's now "exit if data array has < 2 values".

I've done "<2" so that both length 0 and length 1 arrays are captured